### DISCUSSION TITLE
Create Render Event Handling Help

### DISCUSSION THREAD

**Discussion by rgfx** [bryanmylee/svelte-headless-table#85] [Q&A] [answered]
Hi,  I am pretty sure I can't create an Dispatcher with Create Render. So I think best option is create a click event. But I am having troubles getting it work. If you could be so kind as to steer me in the right direction. 

```js
	const handleClick = (ev, value) => {
		console.log('click', ev, value);
	}
                table.column({
			header: "",
			accessor: "id",
			cell: ({ value }) =>
				createRender(EditButton, {
					id: value,
				}).on('click', (ev) => handleClick(ev, value)),
		}),
```		

I created this repl to show my issue. 
https://svelte.dev/repl/dc243ee1ecc4421faa71865992800b9f?version=3.48.0

**Comment by bryanmylee** [comment, marked as answer]
You can use a dispatcher with `createRender`! It will handle all events that the component dispatches, including native and `createEventDispatcher` events.

The issue with your example is that your `EditButton.svelte` isn't dispatching any events.

You can fix it by using Svelte's [event forwarding](https://svelte.dev/tutorial/event-forwarding):

```svelte
<!-- EditButton.svelte -->
<script>
  export let id;
</script>

<button on:click>
  {id}
</button>
```

or using an event dispatcher:

```svelte
<!-- EditButton.svelte -->
<script>
  import {createEventDispatcher} from 'svelte';

  const dispatch = createEventDispatcher();
  export let id;
</script>

<button on:click={() => dispatch('click')}>
  {id}
</button>
```

**Reply by rgfx** [reply]
Thank you, I should of tried that :)
